src/index.js

```diff
@@ -35,43 +36,77 @@ function glamorous(comp, {rootEl, displayName} = {}) {
      * This is a component which will render the comp (closure)
      * with the glamorous styles (closure). Forwards any valid
      * props to the underlying component.
-     * @param {Object} props the props for the component. className is
-     *   handled specially so any glamor-provided classNames will be
-     *   merged predicably (with no regard to specificity)
+     * @param {Object} theme the theme object
      * @return {ReactElement} React.createElement
      */
-    function GlamorousComponent(props) {
-      const {className, ...rest} = props
-      const {toForward, cssOverrides} = splitProps(rest, GlamorousComponent)
-
-      // create className to apply
-      const mappedArgs = GlamorousComponent.styles.map(mapToPropsCall)
-      const {
-        glamorStyles: parentGlamorStyles,
-        glamorlessClassName,
-      } = extractGlamorStyles(className)
-      const glamorClassName = css(
-        ...mappedArgs,
-        ...parentGlamorStyles,
-        cssOverrides,
-      ).toString()
-      const fullClassName = joinClasses(glamorlessClassName, glamorClassName)
-
-      return React.createElement(GlamorousComponent.comp, {
-        className: fullClassName,
-        ...toForward,
-      })
-
-      function mapToPropsCall(glamorRules) {
-        if (typeof glamorRules === 'function') {
-          return glamorRules(props)
+    class GlamorousComponent extends Component {
+      state = {theme: null}
+      setTheme = theme => this.setState({theme})
+
+      componentWillMount() {
+        const {theme} = this.props
+        if (this.context[CHANNEL]) {
+          // if a theme is provided via props, it takes precedence over context
+          this.setTheme(theme ? theme : this.context[CHANNEL].getState())
```

Comment:
If we want to support setting the `theme` to `null` (which we sorta do below), then this should probably be:

```javascript
this.setTheme(theme || this.context[CHANNEL].getState())
```